### PR TITLE
feat: add probability to cards in pack

### PR DESCRIPTION
**Pull request by kurpav**

### CODE REVIEW

File: js/packages/web/src/views/pack/components/RedeemModal/components/RedeemCard/index.tsx [hunk 1]
```diff
@@ -17,9 +21,9 @@ const RedeemCard = ({ item: { info, pubkey } }: IPropsRedeemCard) => (
         <p className="info__title">{info.data.name}</p>
       </div>
     </div>
-    {/* <div className="modal-redeem__percentage">
-      <p>{`${percentage}% chance`}</p>
-    </div> */}
+    <div className="modal-redeem__percentage">
+      <p>{`${probability.toFixed(1)}% chance`}</p>
```
Comment (b2kdaman) [inline js/packages/web/src/views/pack/components/RedeemModal/components/RedeemCard/index.tsx:25]: You sure that probability cant go lower than 0.1?

File: js/packages/web/src/views/pack/components/RedeemModal/hooks/useMetadataByPackCard.ts [hunk 2]
```diff
@@ -23,3 +47,7 @@ export const useMetadataByPackCard = (
 
   return metadata;
 };
+
+function calculateProbability(count: number, total: number): number {
+  return (count * 100) / total;
```
Comment (b2kdaman) [inline js/packages/web/src/views/pack/components/RedeemModal/hooks/useMetadataByPackCard.ts:52]: What if total is 0? How possible is this scenario?

File: js/packages/web/src/views/pack/components/RedeemModal/components/RedeemCard/index.tsx (hunk 1 above)
Comment (kurpav) [reply js/packages/web/src/views/pack/components/RedeemModal/components/RedeemCard/index.tsx:25]: we have only 1 decimal after comma on the designs. I don't know if such a case is possible.
cc @m-sebastiyan

File: js/packages/web/src/views/pack/components/RedeemModal/hooks/useMetadataByPackCard.ts (hunk 2 above)
Comment (kurpav) [reply js/packages/web/src/views/pack/components/RedeemModal/hooks/useMetadataByPackCard.ts:52]: I'd say there are no such cases, but I can return 0 in this case.
cc @m-sebastiyan

File: js/packages/web/src/views/pack/components/RedeemModal/components/RedeemCard/index.tsx (hunk 1 above)
Comment (m-sebastiyan) [reply js/packages/web/src/views/pack/components/RedeemModal/components/RedeemCard/index.tsx:25]: Can you do this in the hook?

File: js/packages/web/src/views/pack/components/RedeemModal/hooks/useMetadataByPackCard.ts [hunk 3]
```diff
@@ -1,19 +1,43 @@
...
     () =>
       cards?.reduce<PackMetadataByPackCard>(
         (packMetadata, { info, pubkey }) => {
-          packMetadata[pubkey] = metadataByMasterEdition[info.master];
+          const probCount = isMaxSupplyDistribution
+            ? info.maxSupply
+            : info.weight;
+          packMetadata[pubkey] = {
+            ...metadataByMasterEdition[info.master],
+            probability: calculateProbability(probCount, total.toNumber()),
```
Comment (m-sebastiyan) [inline js/packages/web/src/views/pack/components/RedeemModal/hooks/useMetadataByPackCard.ts:39]: What about toFixed here?

File: js/packages/web/src/views/pack/components/RedeemModal/hooks/useMetadataByPackCard.ts (hunk 2 above)
Comment (m-sebastiyan) [reply js/packages/web/src/views/pack/components/RedeemModal/hooks/useMetadataByPackCard.ts:52]: Sounds good

**Review by m-sebastiyan — APPROVED** [review]
Nicee, left a couple of comments

**Review by b2kdaman — APPROVED** [review]
lgtm 👍

**Pull request merged**